File: adventOfCode/Day1.cpp
#include "Util.h"

using namespace std;

int main()
{
	//DAY 1: 
	std::vector<unsigned char> buffer;
	if (utility::readFileToBuffer("Day1.txt", buffer)) {
		int total = 0;
		int size = buffer.size();
		int shift = size / 2; // shift = 1 for the first assignment


		for (int i = 0; i<size; i++)
		{
			if ((int)buffer[i] == (int)buffer[(i + shift) % size]) {
				total = total + (buffer[i] - '0');
			}
		}

		cout << total << endl;
		return 0;
	}
	else {
		return 1;
	}

}

File: adventOfCode/Day1.py
import os

def Input():
	"Open input file of corresponding day."
	print os.path.splitext(os.path.basename(__file__))[0]
	with open('{}.txt'.format(os.path.splitext(os.path.basename(__file__))[0]), 'r') as myfile:
		data=myfile.read()
	return data

#Day 1: Inverse Captcha
input = Input()
total = 0
shift = len(input)/2 # for first puzzle, shift = 1
for idx, c in enumerate(input):
	if int(c) == int(input[(idx+shift)%len(input)]):
		total = total + int(c)
print total
